packages/transaction-controller/src/TransactionController.ts

```diff
@@ -1107,6 +1090,86 @@ export class TransactionController extends BaseController<
     return this.nonceTracker.getNonceLock(address);
   }
 
+  private onBootCleanup() {
+    this.initApprovalsOfUnapprovedTransactions();
+    this.loadGasValuesOfUnapprovedTransactions();
+    this.submitApprovedTransactions();
+  }
+
+  /**
+   * Create approvals for all unapproved transactions on current chain.
+   */
+  private initApprovalsOfUnapprovedTransactions() {
+    const chainId = this.getChainId();
+    const unapprovedTransactions = this.state.transactions.filter(
+      (transaction) =>
+        transaction.status === TransactionStatus.unapproved &&
+        transaction.chainId === chainId,
+    );
+
+    for (const transactionMeta of unapprovedTransactions) {
+      this.processApproval(transactionMeta, {
+        shouldShowRequest: false,
+      }).catch((error) => {
+        /* istanbul ignore next */
+        console.error('Error during persisted transaction approval', error);
+      });
+    }
+  }
+
+  /**
+   * Update the gas values of all unapproved transactions on current chain.
+   */
+  private async loadGasValuesOfUnapprovedTransactions() {
+    const chainId = this.getChainId();
+    const isEIP1559Compatible = await this.getEIP1559Compatibility();
+    const unapprovedTransactions = this.state.transactions.filter(
+      (transaction) =>
+        transaction.status === TransactionStatus.unapproved &&
+        transaction.chainId === chainId,
+    );
+
+    for (const transactionMeta of unapprovedTransactions) {
+      try {
+        await updateGasFees({
+          eip1559: isEIP1559Compatible,
+          ethQuery: this.ethQuery,
+          getGasFeeEstimates: this.getGasFeeEstimates.bind(this),
+          txMeta: transactionMeta,
+        });
+        this.updateTransaction(
+          transactionMeta,
+          'TransactionController:loadGasValuesOfUnapprovedTransactions - gas values updated',
+        );
+      } catch (error) {
+        this.failTransaction(transactionMeta, error as Error);
+        /* istanbul ignore next */
+        console.error(
+          'Error during loading gas values for unapproved transactions',
+          error,
+        );
+      }
+    }
+  }
+
+  /**
+   * Force to submit approved transactions on current chain.
+   */
+  private submitApprovedTransactions() {
```

Comment:
Once MMI hooks are merged we need to use the `beforeApproveOnInit` (we can rename it :sweat_smile: ) [here](https://github.com/MetaMask/core/pull/1787/files#diff-96bc10c64242a8e23f5835532988319189239bdf40f556b86199dcbcc41c7bfcR374), so just point it here depending on the timing of those two PR being merge we might need to create a separate ticket to address it.

Reply:
Yes definitely, we can make this work either this or another goes first.